### PR TITLE
Add origin tooling necessary for building

### PR DESCRIPTION
**Pull request by jpeeler**
This is essentially a copy of the hack directory from origin, with necessary changes to make build-go, build-cross, and build-images to work.

### CODE REVIEW

**Comment by deads2k** [comment]
A lot of this looks unnecessary.  dind scripts?  the e2e test calls?

**Comment by jpeeler** [comment]
Definitely some of it is. I didn't have time to prune the hack directory, it was simply copied from origin in its entirety.

File: hack/build-images.sh [hunk 1]
```diff
@@ -109,44 +116,54 @@ function image() {
...
-image "${tag_prefix}-docker-builder"        images/builder/docker/docker-builder
-image "${tag_prefix}-sti-builder"           images/builder/docker/sti-builder
-image "${tag_prefix}-f5-router"             images/router/f5
-image "openshift/node"                      images/node
-# images that depend on "openshift/node"
-image "openshift/openvswitch"               images/openvswitch
-
-# extra images (not part of infrastructure)
-image "openshift/hello-openshift"           examples/hello-openshift
+tag_prefix="${OS_IMAGE_PREFIX:-"openshift/service-catalog"}"
```
Comment (stevekuznetsov) [inline hack/build-images.sh]: I would not have expected this to change

File: build/apiserver/Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM debian
+FROM centos:7
 
 ADD tmp/apiserver /opt/services/
```
Comment (stevekuznetsov) [inline build/apiserver/Dockerfile]: Will you carry patches in this repo to change these to be `yum install` stanzas or will you have two Dockerfiles, one for "dev" that does this and one with `yum`?

File: hack/build-images.sh (hunk 1 above)
Comment (jpeeler) [inline hack/build-images.sh]: I think this makes sense so the image names have zero collision potential.

**Comment by stevekuznetsov** [comment]
> I think this makes sense so the image names have zero collision potential.

The images should still honor the prefix, so use:

```sh
tag_prefix="${OS_IMAGE_PREFIX:-"openshift/origin"}-service-catalog"
```

Instead of 

```sh
tag_prefix="${OS_IMAGE_PREFIX:-"openshift/service-catalog"}"
```

**Comment by stevekuznetsov** [comment]
> The images should still honor the prefix

The reason being that there is a huge load of tooling, etc that expects these prefixes to exist and be consistent (the OCP one is `openshift3/ose`)

File: build/apiserver/Dockerfile (hunk 2 above)
Comment (jpeeler) [inline build/apiserver/Dockerfile]: I've now added Dockerfiles for RPM installation alongside Dockerfile.dev files that copy in the binary.

File: hack/build-images.sh (hunk 1 above)
Comment (jpeeler) [inline hack/build-images.sh]: I made the suggested change of keeping the openshift/origin prefix.

**Comment by pmorie** [comment]
I think we should reset this repo to head of upstream and then make this PR just the additive changes to enable OSBS

**Comment by jpeeler** [comment]
@pmorie as far as I can tell, this is just github not rendering the commits to be merged correctly (perhaps because I force pushed the base branch after making this PR?). The commits before my 5 commits are actually already in master of this repo.

**Comment by pmorie** [comment]
@jpeeler I believe you need to rebase this branch and the commits should render correctly.

**Pull request merged**